mmseg/models/backbones/strideformer.py

```diff
@@ -0,0 +1,852 @@
+        low_feat2 = self.embedding_list[1](low_feat2)
+
+        high_feat_act = F.interpolate(
+            self.act_embedding_list[2](inputs[2]),
+            size=low_feat2.shape[2:],
+            mode='bilinear',
+        )
+        high_feat = F.interpolate(
+            self.embedding_list[2](inputs[2]),
+            size=low_feat2.shape[2:],
+            mode='bilinear')
+
+        res = (
+            low_feat1_act * low_feat2_act * high_feat_act *
+            (low_feat1 + low_feat2) + high_feat)
+
+        return res
+
+
+class InjectionMultiSumallmultiallsumSimpx8(nn.Module):
+
+    def __init__(
+            self,
+            in_channels=(64, 128, 256, 384),
+            act_cfg=None,
+            out_channels=256,
+            norm_cfg=dict(type='BN'),
+    ):
+        super().__init__()
+        self.embedding_list = nn.ModuleList()
+        self.act_embedding_list = nn.ModuleList()
+        self.act_list = nn.ModuleList()
+        for i in range(len(in_channels)):
+            if i != 1:
+                self.embedding_list.append(
+                    ConvModule(
+                        in_channels=in_channels[i],
+                        out_channels=out_channels,
+                        kernel_size=1,
+                        bias=False,
+                        norm_cfg=norm_cfg,
+                        act_cfg=None,
+                    ))
+            if i != 0:
+                self.act_embedding_list.append(
+                    ConvModule(
+                        in_channels=in_channels[i],
+                        out_channels=out_channels,
+                        kernel_size=1,
+                        bias=False,
+                        norm_cfg=norm_cfg,
+                        act_cfg=act_cfg,
+                    ))
+
+    def forward(self, inputs):
+        # x_x8, x_x16, x_x32
+        low_feat1 = self.embedding_list[0](inputs[0])
+
+        low_feat2 = F.interpolate(
+            inputs[1], size=low_feat1.shape[-2:], mode='bilinear')
+        low_feat2_act = self.act_embedding_list[0](low_feat2)
+
+        high_feat_act = F.interpolate(
+            self.act_embedding_list[1](inputs[2]),
+            size=low_feat2.shape[2:],
+            mode='bilinear',
+        )
+        high_feat = F.interpolate(
+            self.embedding_list[1](inputs[2]),
+            size=low_feat2.shape[2:],
+            mode='bilinear')
+
+        res = low_feat2_act * high_feat_act * low_feat1 + high_feat
+
+        return res
+
+
+def _make_divisible(v, divisor=8, min_value=None):
+    if min_value is None:
+        min_value = divisor
+    new_v = max(min_value, int(v + divisor / 2) // divisor * divisor)
+    if new_v < 0.9 * v:
+        new_v += divisor
+    return new_v
+
+
+@MODELS.register_module()
+class HSigmoid(nn.Module):
+
+    def __init__(self):
+        super().__init__()
+        self.relu = nn.ReLU6()
+
+    def forward(self, x):
+        return self.relu(x + 3) / 6
+
```

Comment:
Might use the implementation in mmcv https://github.com/open-mmlab/mmcv/blob/main/mmcv/cnn/bricks/hsigmoid.py#L10